engine/src/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java

```diff
@@ -89,8 +90,13 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
     if ( first ) {
 
       first = false;
-      data.outputRowMeta = getInputRowMeta().clone();
-      data.inputRowMeta = getInputRowMeta().clone();
+      if ( r == null ) {
```

Comment:
I would agree for most other steps, but there is some functionality that would allow for an "empty" (only header row) file to be created, even if no data rows are received by the step.

By placing this here, instead, it will allow the empty file to be created, if configured, and then drop down to line 197/203 to close out the file.

There's still an integration test needed for this use-case, which I'll add shortly.